File: app/src/main/java/id/indosw/xlstojson/jsonstringtoxls/ConvertJsonToExcel.java
package id.indosw.xlstojson.jsonstringtoxls;

import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.fasterxml.jackson.databind.ObjectMapper;

import org.apache.poi.ss.usermodel.Cell;
import org.apache.poi.ss.usermodel.CellStyle;
import org.apache.poi.ss.usermodel.CreationHelper;
import org.apache.poi.ss.usermodel.Font;
import org.apache.poi.ss.usermodel.IndexedColors;
import org.apache.poi.ss.usermodel.Row;
import org.apache.poi.ss.usermodel.Sheet;
import org.apache.poi.ss.usermodel.Workbook;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import java.io.FileOutputStream;
import java.io.IOException;
import java.util.List;

@SuppressWarnings("SpellCheckingInspection")
public class ConvertJsonToExcel {

    @SuppressWarnings("TryWithIdenticalCatches")
    public static List<Customer> convertJsonString2Objects(String jsonString){
        List<Customer> customers = null;

        try {
            customers = new ObjectMapper().readValue(jsonString, new TypeReference<List<Customer>>(){});
        } catch (JsonParseException e) {
            e.printStackTrace();
        } catch (JsonMappingException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

        return customers;
    }

    public static void writeObjects2ExcelFile(List<Customer> customers, String filePath) throws IOException {
        String[] COLUMNs = {"Id", "Name", "Address", "Age"};
        Workbook workbook = new XSSFWorkbook();
        CreationHelper createHelper = workbook.getCreationHelper();
        Sheet sheet = workbook.createSheet("Customers");
        Font headerFont = workbook.createFont();
        headerFont.setBold(true);
        headerFont.setColor(IndexedColors.BLUE.getIndex());
        CellStyle headerCellStyle = workbook.createCellStyle();
        headerCellStyle.setFont(headerFont);
        // Row for Header
        Row headerRow = sheet.createRow(0);
        // Header
        for (int col = 0; col < COLUMNs.length; col++) {
            Cell cell = headerRow.createCell(col);
            cell.setCellValue(COLUMNs[col]);
            cell.setCellStyle(headerCellStyle);
        }
        // CellStyle for Age
        CellStyle ageCellStyle = workbook.createCellStyle();
        ageCellStyle.setDataFormat(createHelper.createDataFormat().getFormat("#"));

        int rowIdx = 1;
        for (Customer customer : customers) {
            Row row = sheet.createRow(rowIdx++);

            row.createCell(0).setCellValue(customer.getId());
            row.createCell(1).setCellValue(customer.getName());
            row.createCell(2).setCellValue(customer.getAddress());

            Cell ageCell = row.createCell(3);
            ageCell.setCellValue(customer.getAge());
            ageCell.setCellStyle(ageCellStyle);
        }

        FileOutputStream fileOut = new FileOutputStream(filePath);
        workbook.write(fileOut);
        fileOut.close();
        workbook.close();
    }
}

File: app/src/main/java/id/indosw/xlstojson/kotlin/jsonstringtoxls/ConvertJsonToExcel.kt
package id.indosw.xlstojson.kotlin.jsonstringtoxls

import com.fasterxml.jackson.core.JsonParseException
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.JsonMappingException
import com.fasterxml.jackson.databind.ObjectMapper
import org.apache.poi.ss.usermodel.IndexedColors
import org.apache.poi.ss.usermodel.Workbook
import org.apache.poi.xssf.usermodel.XSSFWorkbook
import java.io.FileOutputStream
import java.io.IOException

@Suppress("LocalVariableName", "SpellCheckingInspection")
object ConvertJsonToExcel {
    @Suppress("LocalVariableName")
    fun convertJsonString2Objects(jsonString: String?): List<Customer?>? {
        var customers: List<Customer?>? = null
        @Suppress("LocalVariableName")
        try {
            customers =
                ObjectMapper().readValue(jsonString, object : TypeReference<List<Customer?>?>() {})
        } catch (e: JsonParseException) {
            e.printStackTrace()
        } catch (e: JsonMappingException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return customers
    }

    @Throws(IOException::class)
    fun writeObjects2ExcelFile(customers: List<Customer>, filePath: String?) {
        val COLUMNs = arrayOf("Id", "Name", "Address", "Age")
        val workbook: Workbook = XSSFWorkbook()
        val createHelper = workbook.creationHelper
        val sheet = workbook.createSheet("Customers")
        val headerFont = workbook.createFont()
        headerFont.bold = true
        headerFont.color = IndexedColors.BLUE.getIndex()
        val headerCellStyle = workbook.createCellStyle()
        headerCellStyle.setFont(headerFont)
        // Row for Header
        val headerRow = sheet.createRow(0)
        // Header
        for (col in COLUMNs.indices) {
            val cell = headerRow.createCell(col)
            cell.setCellValue(COLUMNs[col])
            cell.cellStyle = headerCellStyle
        }
        // CellStyle for Age
        val ageCellStyle = workbook.createCellStyle()
        ageCellStyle.dataFormat = createHelper.createDataFormat().getFormat("#")
        var rowIdx = 1
        for (customer in customers) {
            val row = sheet.createRow(rowIdx++)
            row.createCell(0).setCellValue(customer.id)
            row.createCell(1).setCellValue(customer.name)
            row.createCell(2).setCellValue(customer.address)
            val ageCell = row.createCell(3)
            ageCell.setCellValue(customer.age.toDouble())
            ageCell.cellStyle = ageCellStyle
        }
        val fileOut = FileOutputStream(filePath)
        workbook.write(fileOut)
        fileOut.close()
        workbook.close()
    }
}